src/components/MyrTour.js

```diff
@@ -20,8 +25,13 @@ class MyrTour extends Component {
         <Tour
           steps={steps}
           maskClassName="mask"
-          isOpen={this.state.isTourOpen}
-          onAfterOpen={this.props.isEditorClosed ? this.props.toggleEditor : ()=>{}}
+          isOpen={this.state.isTourOpen}  
+          onAfterOpen={()=>{
+            this.viewOnlyOnOpen = this.props.viewOnly;
```

Comment:
What are you trying to do with this line here? If you're trying to set the state, you need to use `this.setState({viewOnlyOnOpen: this.props.viewOnly})`. However, you did already set that in the constructor.

Reply:
i realized the tour object was created when the page loads, rather than when the tour button is clicked. since the editor is always open by default when the page loads, viewOnlyOnOpen was always evaluating to false b/c its value was only set when the tour object was created. this line checks if the editor is open or not and updates viewOnlyOnOpen to match. it would probably be better off in the component state anyway though, so i will move it there.